stl/inc/future

```diff
@@ -170,7 +172,7 @@ class _Associated_state;
 
 template <class _Ty>
 struct __declspec(novtable) _Deleter_base { // abstract base class for managing deletion of state objects
-    virtual void _Delete(_Associated_state<_Ty>*) = 0;
+    virtual void _Delete(_Associated_state<_Ty>*) noexcept = 0;
     virtual ~_Deleter_base() noexcept {}

```

Comment:
While we're here, we may as well:
```suggestion
    virtual ~_Deleter_base() = default;
```

Reply:
I'm not sure whether it would be better to make this dtor non-virtual (and implicitly declared), because this dtor seems never virtually called. Perhaps such change should be done in vNext?

Reply:
`protected` and non-`virtual` _would_ be better, but IMO has no benefits commensurate with the risks of ABI issues, especially in CLR metadata. We could add a `// TRANSITION, vNext (make non-virtual, never called polymorphically)` and/or add a bullet to the "things to change in vNext" issue.

Reply:
I just saw https://github.com/microsoft/STL/issues/208#issuecomment-1350452371 which records this suggestion, so I think my "add a comment" suggestion is unnecessary.